src/builder/Build.kt

```diff
@@ -22,21 +24,21 @@ class Build(val game: Game, val global: Global) {
         var reward = 250
         if (currentLevel > 1) {
             for (i in 1 until (currentLevel)) {
-                reward += game.waves[i]!!.totalReward
+                reward += game.data.waves[i]!!.totalReward
                 reward += lane.getIncome(i)
             }
         }
         return reward
     }
 
     fun levelIncrease() {
-        if (game.waves.containsKey(currentLevel + 1)) {
+        if (currentLevel < game.data.waves.size) {
             currentLevel++
         }
     }
 
     fun levelDecrease() {
-        if (game.waves.containsKey(currentLevel - 1)) {
+        if (currentLevel > 0) {

```

Comment:
should be `> 1`? or currentLevel should be initialized with 0 and when rendering add +1